Runtime/ReorderableArray.cs

```diff
@@ -6,7 +6,7 @@
 namespace Malee.List {
 
 	[Serializable]
-	public abstract class ReorderableArray<T> : ICloneable, IList<T>, ICollection<T>, IEnumerable<T> {
+	public abstract class ReorderableArray<T> : ICloneable, IList<T>, ICollection<T>, IEnumerable<T>, IReadOnlyList<T> {
```

Comment:
IReadOnlyList<T> makes ICollection<T> and IEnumerable<T> redundantly defined.

Reply:
Though it is a tiny bit redundant, I feel that implementing `IReadOnlyList` and leaving the `IEnumerable` declaration is still helpful. It's hard for developers to memorize which interfaces implement what. So leaving the interfaces explicitly declared is usually easier to read IMHO. Same goes for the `ICollection` interface.